io.openems.edge.evcs.ocpp.server/src/io/openems/edge/evcs/ocpp/server/MyJsonServer.java

```diff
@@ -122,10 +139,16 @@ public void lostSession(UUID sessionIndex) {
 				for (Entry<String, UUID> session : MyJsonServer.this.parent.ocppSessions.entrySet()) {
```

Comment:
To address the issue mentioned below, we added the following changes. However, these modifications might not be necessary if our only concern is the brief window between establishing the new session and discarding the old one, as they make the code more complex.

> When the old connection is disconnected, it should ensure that the lostSession for the charge point referenced by the new connection is not invoked.

```java
        if (sessionEvcss != null) {
          for (AbstractManagedOcppEvcsComponent ocppEvcs : sessionEvcss) {
            var ocppId = ocppEvcs.getConfiguredOcppId();
            UUID latest = MyJsonServer.this.parent.ocppSessions.get(ocppId);
            /**
             * Since multiple sessions are connected with the same ocppId, if
             * `lostSession` is executed with an old session, the `sessionId` of the
             * component corresponding to the latest session would be set to null.
             */
            if (latest != sessionIndex) {
              MyJsonServer.this.logWarn("skip lostSession of old session [" + sessionIndex +
                  "] for ocppIdentifier [" + ocppId + "]");
              continue;
            }
            ocppEvcs.lostSession();
            MyJsonServer.this.logWarn("lostSession [" + sessionIndex + "] for ocppIdentifier [" + ocppId + "]");
          }
          MyJsonServer.this.parent.activeEvcsSessions.remove(sessionIndex);
          MyJsonServer.this.logWarn("remove activeEvcsSessions [" + sessionIndex + "]");
        }
```